root/var/www/html/freepbx/rest/modules/configuration.php

```diff
@@ -141,3 +141,44 @@ function setLegacyMode($value) {
     }
 });
 
+/*
+* GET /configuration/externalip
+*/
+$app->get('/configuration/externalip', function (Request $request, Response $response, $args) {
+    return $response->withJson(\FreePBX::create()->Sipsettings->getConfig('externip'),200);
+});
+
+/*
+* POST /configuration/externalip/{ip}
+*/
+$app->post('/configuration/externalip/{ip}', function (Request $request, Response $response, $args) {
+    $route = $request->getAttribute('route');
+    $ip = $route->getArgument('ip');
+    if (\FreePBX::create()->Sipsettings->setConfig('externip',$ip)) {
+        system('/var/www/html/freepbx/rest/lib/retrieveHelper.sh > /dev/null &');
+        return $response->withStatus(200);
+    }
+    return $response->withStatus(500);
+});
+
+/*
+* GET /configuration/suggestedip
+*/
+$app->get('/configuration/suggestedip', function (Request $request, Response $response, $args) {
+    $ch = curl_init();
+    curl_setopt($ch, CURLOPT_URL, "http://ifconfig.io/ip");
+    curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
+    curl_setopt($ch, CURLOPT_TIMEOUT, 4);
+    curl_setopt($ch, CURLOPT_HTTPHEADER, array(
+        "Content-Type: application/json;charset=utf-8",
+        "Accept: application/json;charset=utf-8",
+    ));
```

Comment:
The above headers are misleading: the ifconfig.io services ignores them and return a text/plain payload.

I suppose the withJson() returns a JSON string, whilst I was expecting a JSON object with some attributes...

Reply:
I removed the header, but don't the second part